BWEM/src/area.cpp

```diff
@@ -414,6 +413,138 @@ void Area::CreateBases()
 	}
 }
 
+// Thanks to Dave Churchill and the SAIDA team
+void Area::calcBoundaryVertices() const
+{
+	set<Position> unsortedVertices;
+
+	for (auto x(0); x < Map::Instance().Size().x; ++x)
+		for (auto y(0); y < Map::Instance().Size().y; ++y)
+		{
+			TilePosition tp{ x, y };
+
+			if (Map::Instance().GetArea(tp) != this)
+			{
+				continue;
+			}
+
+			TilePosition t1{ tp.x + 1, tp.y };
+			TilePosition t2{ tp.x, tp.y + 1 };
+			TilePosition t3{ tp.x - 1, tp.y };
+			TilePosition t4{ tp.x, tp.y - 1 };
+
+			// a tile is 'surrounded' if
+			// 1) in all 4 directions there's a tile position in the current region
+			// 2) in all 4 directions there's a buildable tile
+			auto surrounded = true;
+
+			if (tp.x == 0 || tp.y == 0
+				|| (t1.isValid() && (Map::Instance().GetArea(t1) != this || !Broodwar->isBuildable(t1)))
+				|| (t2.isValid() && (Map::Instance().GetArea(t2) != this || !Broodwar->isBuildable(t2)))
+				|| (t3.isValid() && (Map::Instance().GetArea(t3) != this || !Broodwar->isBuildable(t3)))
+				|| (t4.isValid() && (Map::Instance().GetArea(t4) != this || !Broodwar->isBuildable(t4))))
+			{
+				surrounded = false;
+			}
+
+			// Area
+			// push the tiles that aren't surrounded
+			if (!surrounded && Broodwar->isBuildable(tp))
+			{
+				unsortedVertices.insert(Position{ tp } +Position{ 16, 16 });
+			}
+		}
+
+	//printf("Area ID = %d, # of unsortedVertices = %d\n", Id(), unsortedVertices.size());
+
+	if (unsortedVertices.empty()) {
+		return;
+	}
+
+	vector<Position> sortedVertices;
+	auto current = *unsortedVertices.begin();
+
+	sortedVertices.push_back(current);
+	unsortedVertices.erase(current);
+
+	// while we still have unsorted vertices left, find the closest one remaining to current
+	while (!unsortedVertices.empty())
```

Comment:
Instead of doing this entire thing, could we skip the `std::set` and just use a single vector, sorting the `[1:n]` elements?